src/Client/LocalConnection.cpp

```diff
@@ -356,7 +356,23 @@ bool LocalConnection::poll(size_t)
 
         try
         {
-            while (pollImpl());
+            while (pollImpl())
+            {
+                LOG_DEBUG(&Poco::Logger::get("LocalConnection"), "Executor timeout encountered");
+
+                if (send_progress && (state->after_send_progress.elapsedMicroseconds() >= query_context->getSettingsRef().interactive_delay))
```

Comment:
Would be good to unite the code in lines 344-355  into a separate method may be - to avoid copy pasted code, may be something like 
```
if (needSendProgressOrMetrics)
    return true;
```
where 
```
bool needSendProgressOrMetrics()
{
    if (send_progress && (state->after_send_progress.elapsedMicroseconds() >= query_context->getSettingsRef().interactive_delay))
    {
            state->after_send_progress.restart();
            next_packet_type = Protocol::Server::Progress;
            return true;
    }
    if (send_profile_events && (state->after_send_profile_events.elapsedMicroseconds() >= query_context->getSettingsRef().interactive_delay))
     {
            sendProfileEvents();
            return true;
    }
    return false;
}
```
Though this is just a suggestion, not necessary